src/app/Http/Controllers/Operations/InlineCreateOperation.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Backpack\CRUD\app\Http\Controllers\Operations;
+
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Illuminate\Support\Facades\Route;
+
+trait InlineCreateOperation
+{
+    /**
+     * Define which routes are needed for this operation.
+     *
+     * @param string $segment    Name of the current entity (singular). Used as first URL segment.
+     * @param string $routeName  Prefix of the route name.
+     * @param string $controller Name of the current CrudController.
+     */
+    protected function setupInlineCreateOperationRoutes($segment, $routeName, $controller)
+    {
+        Route::get($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create',
+            'uses'      => $controller.'@getInlineCreateModal',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::post($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create-save',
+            'uses'      => $controller.'@storeInlineCreate',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::get($segment.'/inline/refresh', [
+            'as'        => $segment.'-inline-refresh-options',
+            'uses'      => $controller.'@inlineRefreshOptions',
+            'operation' => 'InlineCreateOperation',
+        ]);
+    }
+
+    public function setupInlineCreateDefaults()
+    {
+        $this->crud->setOperationSetting('inline_create', true);
+    }
+
+    public function getInlineCreateModal()
```

Comment:
Sorry, i am just used to write docs in the end. I knew beforehand that this would change alot. But you got the point, for code review it's wayyyy easier. 

Thanks for the excelent tips you give me in your reviews :+1:

Reply:
Oh yeah, you're write. And with the amount of rewrites, you're totally right. Best to write the docblocks at the very end 😄